### PR TITLE
Feat: Create favorite event route

### PR DESCRIPTION
**Pull request by josenriagu**
# Description
This Pull Request creates and implements the favorite event route
## Type of change
- [x] New feature (non-breaking change which adds functionality)
## Change status
- [x] Complete, tested, ready to review and merge
# How Has This Been Tested?
Please describe steps taken to ensure this feature is functional and does not break other features:
# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by josenriagu** [comment]
[User can like event](https://trello.com/c/e6omwLd7/58-user-can-like-event)

File: resources/favEvents/favEvents.controller.js [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+      evArr.map((event) => Event.findOne({
+        scrapedEventId: event.eventId,
+      }, (err, doc) => {
+        if (err) return false;
+        return doc._doc;
+      })),
+    )
+      .then((modelArr) => modelArr.map((model) => model._doc));
+    return result;
+  },
```
Comment (josenriagu) [inline resources/favEvents/favEvents.controller.js]: @abidex4yemi @twumm @theallegrarr Do you think the efficiency of this findEvent controller will be optimal when the database becomes massive and the number of documents to be traversed increases?

Comment (abidex4yemi) [reply resources/favEvents/favEvents.controller.js]: We would definitely refactor this in RC3, thanks for the feedback

File: resources/favEvents/favEvents.route.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const express = require('express');
+
+const router = express.Router();
+
+const favEventsController = require('./favEvents.controller');
+
+router.get('/', async (req, res) => {
+  try {
+    const { u } = req.query;
```
Comment (abidex4yemi) [inline resources/favEvents/favEvents.route.js]: Please use a good naming convention, `u` to me is not intuitive

File: resources/favEvents/favEvents.route.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if (typeof u === 'undefined') {
+      res.status(400).json('Please supply valid id in the query');
+    } else {
+      const favEventsId = await favEventsController.findFav(u);
+      if (!favEventsId) {
+        throw new Error('Favorite events could not be retrieved at this time. Please try again later');
```
Comment (abidex4yemi) [inline resources/favEvents/favEvents.route.js]: Throwing new error doesn't tell the user anything about their error, this should have a response of the error or what happen

**Review by abidex4yemi — CHANGES_REQUESTED** [review]
Good job on this, please remove all `throw Error` because this brings the code block to an alt and does not send response back to the user about the current state of error and why it occur.

File: resources/favEvents/favEvents.route.js (hunk 3 above)
Comment (josenriagu) [reply resources/favEvents/favEvents.route.js]: Alright mate

File: resources/favEvents/favEvents.route.js (hunk 2 above)
Comment (josenriagu) [reply resources/favEvents/favEvents.route.js]: Alright.

**Review by abidex4yemi — APPROVED** [review]
Looks good

**Pull request merged**